rust/signed_doc/src/metadata/content_type.rs

```diff
@@ -94,61 +132,88 @@ impl minicbor::Decode<'_, ()> for ContentType {
         _ctx: &mut (),
     ) -> Result<Self, minicbor::decode::Error> {
         let p = d.position();
-        match d.int() {
-            // CoAP Content Format JSON
-            Ok(val) if val == minicbor::data::Int::from(50_u8) => Ok(Self::Json),
-            // CoAP Content Format CBOR
-            Ok(val) if val == minicbor::data::Int::from(60_u8) => Ok(Self::Cbor),
-            Ok(val) => {
-                Err(minicbor::decode::Error::message(format!(
-                    "unsupported CoAP Content Formats value: {val}"
-                )))
-            },
-            Err(_) => {
-                d.set_position(p);
-                d.str()?.parse().map_err(minicbor::decode::Error::message)
-            },
+        if let Ok(val) = d.int() {
+            let val: u64 = val.try_into().map_err(minicbor::decode::Error::custom)?;
+            Self::try_from(val).map_err(minicbor::decode::Error::message)
+        } else {
+            d.set_position(p);
+            d.str()?.parse().map_err(minicbor::decode::Error::message)
         }
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use minicbor::{Decode, Decoder, Encoder};
+
     use super::*;
 
     #[test]
     fn content_type_string_test() {
-        assert_eq!(
-            ContentType::from_str("application/cbor").unwrap(),
-            ContentType::Cbor
-        );
-        assert_eq!(
-            ContentType::from_str("application/cddl").unwrap(),
-            ContentType::Cddl
-        );
-        assert_eq!(
-            ContentType::from_str("application/json").unwrap(),
-            ContentType::Json
-        );
-        assert_eq!(
-            ContentType::from_str("application/json+schema").unwrap(),
-            ContentType::JsonSchema
-        );
-        assert_eq!(
-            "application/cbor".parse::<ContentType>().unwrap(),
-            ContentType::Cbor
-        );
-        assert_eq!(
-            "application/cddl".parse::<ContentType>().unwrap(),
-            ContentType::Cddl
-        );
-        assert_eq!(
-            "application/json".parse::<ContentType>().unwrap(),
-            ContentType::Json
-        );
-        assert_eq!(
-            "application/json+schema".parse::<ContentType>().unwrap(),
-            ContentType::JsonSchema
-        );
+        let entries = [
```

Comment:
you could replace it with using `test_case` library, I think it would be a great place to do it